src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchAgent.java

```diff
@@ -88,4 +91,37 @@ private static void redefineEdition(Instrumentation instrumentation) throws Clas
         instrumentation.retransformClasses(Class.forName(targetClassName));
     }
 
+    private static class MethodDetails {
+
+        private final String className;
+        private final String methodName;
+        private final String methodBody;
+
+        public MethodDetails(String className, String methodName, String methodBody) {
+            this.className = className;
+            this.methodName = methodName;
+            this.methodBody = methodBody;
+        }
+
+        public String getClassName() {
+            return className;
+        }
+
+        public String getMethodName() {
+            return methodName;
+        }
+
+        public String getMethodBody() {
+            return methodBody;
+        }
+    }
+
+    private enum Edition {
+        CE, WEB;
+
+        private static Optional<Edition> fromString(String input) {
+            return Arrays.stream(values()).filter(v -> v.name().toLowerCase(Locale.ENGLISH).equals(input)).findFirst();
```

Comment:
Will `input` ever come non-lower-cased? 
Does it make sense to first lowercase it before performing lookup?
```suggestion
      return Optional.ofNullable(input)
                     .map(i -> i.toLowerCase(Locale.ENGLISH))
                     .flatMap(i -> Arrays.stream(values())
                                         .filter(v -> v.name()
                                                       .toLowerCase(Locale.ENGLISH)
                                                       .equals(i))
                                         .findFirst());
```

Reply:
The installation instructions only have the values specified in lower case and startup will fail-fast if the values don't match so I'm not too concerned about handling mixed-case. If this does become an issue longer-term then this can be simplified to the following to match case insentiviely:
```java
        return Arrays.stream(values).filter(v -> v.name().equalsIgnoreCase(input)).findFirst();
```